### ISSUE TITLE
Add keyboard shortcuts

### ISSUE THREAD

**Issue by adomasven** [zotero/zotero-connectors#60]

**Comment by hughwilliams94** [comment]
+1 Would be a useful addition

**Comment by adomasven** [comment]
There is no native support for this in Safari and we have _some_ support in BrowserExt as reported above (although doesn't seem to work on MacOS). Either way, we [now have](https://github.com/adomasven/zotero-connectors/blob/e84e1f93cb771c4c39ea9773113083e1b2d7fd28/src/common/inject/inject.jsx#L523) `Zotero.Inject.addKeyboardShortcut()` which works on all browsers.

**Comment by cmcaine** [comment]
Looks like that function never gets called.

You could also use the commands webextension api.

Do you want a PR for this functionality (I'll just add a call to this function at the end of the content script)?

You may also want to consider if you want page content's synthetic events to trigger your shortcut or not.

I wrote a keyboard focused add-on to firefox (https://github.com/tridactyl/tridactyl) and I would like to be able to call zotero from it, so if you do choose to stop page content from triggering your function, would you be willing to expose a function to other add-ons for this purpose? I can help you with this.

**Comment by adomasven** [comment]
We're possibly not as concerned with Safari anymore, given the [uncertain future](https://www.zotero.org/support/kb/safari_12_connector). As for webextensions, the shortcut is defined in the [manifest.json](https://github.com/zotero/zotero-connectors/blob/master/src/browserExt/manifest.json#L52-L58), althought after having looked at the [docs](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/commands) again, we should define a separate shortcut for macOS. We use `Zotero.Inject.addKeyboardShortcut()` for Google Docs integration shortcuts and nothing else right now, nor am I sure we want to, unless necessary.

>I wrote a keyboard focused add-on to firefox (https://github.com/tridactyl/tridactyl) and I would like to be able to call zotero from it, so if you do choose to stop page content from triggering your function, would you be willing to expose a function to other add-ons for this purpose? I can help you with this.

I assume you're talking about `chrome.runtime.onMessageExternal`. We could accept a PR for a change in [background.js](https://github.com/zotero/zotero-connectors/blob/66094fcba6a7709a9ca25587336449ebe6ec43c2/src/browserExt/background.js#L818). How are you (if you are) controlling other extensions though? Do other popular webextensions have APIs to trigger certain functions?

**Comment by cmcaine** [comment]
Ah, I think Ctrl+Shift+S is defined by Firefox on Linux, maybe? Zotero
connector doesn't do anything when I press those keys, anyway.

Tidactyl doesn't yet control any other extensions. We suggested doing so
for ublock and treestyletabs, but neither has happened yet. We just ended
up implementing similar functionality in tridactyl.

The ublock maintainer did reply and was unwilling to provide an API on
first ask, but I never followed up.

Anyway, yes. Such an API would be thru onmessageexternal, probably passing
a tab id unless your code works better with a URL or expects to operate on
the active tab.

If there were any security concerns we could work through those and
possible add a whitelist of other addons that can use the API.
